apps/aecore/src/aec_dry_run.erl

```diff
@@ -96,20 +96,23 @@ dry_run_int([], _Trees, Env, _Opts, Acc) ->
 dry_run_int([{tx, TxOpts, Tx} | Txs], Trees, Env, Opts, Acc) ->
     Stateless = proplists:get_value(stateless, TxOpts, false),
     Env1 = prepare_env(Env, TxOpts),
+    EventsEnabled = proplists:get_bool(tx_events, Opts),
     %% GH3283: Here we should collect and present the `internal_call_tx` events.
     %% This means expanding the return type, and breaking the api :scream_cat:.
     case aec_trees:apply_txs_on_state_trees([Tx], Trees, Env1, [strict, dont_verify_signature|Opts]) of
         {ok, [Tx], [], Trees1, Events} when Stateless ->
-            Env2 = aetx_env:set_events(Env, Events),
-            dry_run_int(Txs, Trees, Env2, Opts, [dry_run_res(Tx, Trees1, ok) | Acc]);
+            OrderedEvents = lists:reverse(Events),
```

Comment:
Wouldn't it make sense to always deliver these in the correct order from `aec_trees`?

Reply:
Yep. Done.